WL-3675 Checks if the email address already exists . If the userId is different from email address

surya-janani opened the pull request:
it sends out validation link.

surya-janani commented:
Calling overloaded method

buckett commented on user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java:
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
+					validationLogic.createValidationAccount(edit.getId(), ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE,valueEmail);
+					addAlert(state,"A validation link has been sent to  "+valueEmail+" Please validate to update your UserId");
+				}
+					UserDirectoryService.commitEdit(edit);
```
Indentation.

buckett commented on user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java:
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
 			{
-				UserDirectoryService.commitEdit(edit);
+				String valueEmail = (String)state.getAttribute("valueEmail");
+				if(EmailValidator.getInstance().isValid(edit.getEid())&& !edit.getEid().equalsIgnoreCase(valueEmail)) {
```
Formatting doesn't match surrounding code.

buckett commented on user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java:
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
+				String valueEmail = (String)state.getAttribute("valueEmail");
+				if(EmailValidator.getInstance().isValid(edit.getEid())&& !edit.getEid().equalsIgnoreCase(valueEmail)) {
+					validationLogic.createValidationAccount(edit.getId(), ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE,valueEmail);
+					addAlert(state,"A validation link has been sent to  "+valueEmail+" Please validate to update your UserId");
```
Needs internationalizations.

buckett commented on user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java:
```diff
@@ -102,6 +105,11 @@
 	private static final String IMPORT_EMAIL="email";
 	private static final String IMPORT_PASSWORD="password";
 	private static final String IMPORT_TYPE="type";
+	ValidationLogic validationLogic;
```
Why have this package visibility and not private?

buckett commented on user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java:
```diff
@@ -1096,9 +1109,20 @@ else if ((mode != null) && (mode.equalsIgnoreCase("edit")) && (!singleUser))
+		if(email != null & !(email.equalsIgnoreCase(user.getEid()))) {
+			try {
+				User userExist = UserDirectoryService.getUserByEid(email);
+				if(userExist != null) {
```
It never returns null so you don't need to check it.

surya-janani commented:
Made review changes. Yet to rebase

buckett commented:
This doesn't merge,

ouit0408 committed:
WL-3675 Checks if the email address already exists . If the userId is different from email address

sends out validation link.

Make validation changes only if the userid is an email acounts
and adding maven changes to add dependency on validation logic

Invoking the overloaded method from validation logic
